src/app/beer_garden/events/events_manager.py

```diff
@@ -1,59 +1,107 @@
-            event.payload = str(ex)
-            raise
         else:
-            if event.name in (
-                Events.INSTANCE_INITIALIZED.name,
-                Events.INSTANCE_STARTED.name,
-                Events.INSTANCE_STOPPED.name,
-                Events.INSTANCE_UPDATED.name,
-                Events.REQUEST_CREATED.name,
-                Events.REQUEST_STARTED.name,
-                Events.REQUEST_COMPLETED.name,
-                Events.SYSTEM_CREATED.name,
-            ):
-                event.payload = result
-            elif event.name in (
-                Events.REQUEST_UPDATED.name,
-                Events.SYSTEM_UPDATED.name,
-            ):
-                event.payload = result
+            if event.name in (Events.REQUEST_UPDATED.name, Events.SYSTEM_UPDATED.name):
                 event.metadata = args[1]
             elif event.name in (Events.QUEUE_CLEARED.name, Events.SYSTEM_REMOVED.name):
                 event.payload = {"id": args[0]}
-            elif event.name in (Events.DB_CREATE.name, Events.DB_UPDATE.name):
-                event.payload = result
             elif event.name in (Events.DB_DELETE.name,):
                 event.payload = args[0]
-        finally:
-            upstream_queue.put(event)
 
-        return result
+        self.do_publish(event)
+
+
+class MainEventManager(EventManagerBase):
+    """Will be used to appropriately route events.
+
+    This will be the main process's manager.
+    """
+
+    def __init__(self):
+        self.forward_queues = []
+        self.callbacks = {
+            Events.INSTANCE_STOPPED.name: [
+                lambda x: print("Instance Stopped", file=sys.stderr)
+            ]
+        }
+
+    def register_forward(self, new_queue):
+        self.forward_queues.append(new_queue)
+
+    def register_callback(self, event_name: str, callback: Callable):
+        if event_name not in self.callbacks:
+            self.callbacks[event_name] = []
+        self.callbacks[event_name].append(callback)
+
+    def process_event(self, event):
+        # First fire any callbacks
+        if event.name in self.callbacks:
+            for callback in self.callbacks[event.name]:
+                callback(event)
+
+        # Then forward to any registered queues
+        for queue in self.forward_queues:
+            queue.put(event)
+
+    def do_publish(self, event):
+        pass
+
+
+class EntryPointManager(EventManagerBase):
+    """Will be used to manage events from an entry point perspective
+
+    This basically ships event up to the main process.
+    """
+
+    def __init__(self, conn):
+        self.conn = conn
+
+    def do_publish(self, event):
+        self.conn.send(event)
+
+
+def publish_event(event_type):
+    @wrapt.decorator
+    def wrapper(wrapped, _, args, kwargs):
+        try:
+            result = wrapped(*args, **kwargs)
+
+            manager.create_event(event_type.name, result, False, args, kwargs)
+
+            return result
```

Comment:
We were talking about this earlier and what values should be forwarded in an event. What you have here should be enough to derive any follow on action. If the event is a delete, then we can just look at the args to get the ID. I think adding any more variables to this would be overkill.